src/ConnectivityWizard/ConnectivityWizard.ts

```diff
@@ -0,0 +1,361 @@
+                scanId: scanId
+            };
+            await next();
+        });
+
+        router.post('/api/finish', async () => {
+            logger.debug('Finish wizard, terminating the process');
+            logger.log('A Repeater has been set up successfully on this machine, please keep this console window open to keep the Repeater running.');
+            
+            process.exit(0);
+        });
+
+        const staticPath: string = process.cwd() + '/wizard-dist/Wizard';
+        logger.debug(staticPath);
+        this.app.use(serve(staticPath));
+        this.app.use(json());
+        this.app.use(bodyParser());
+        this.app.use(router.routes());
+
+        //select available port and launch http listener
+        (async () => {
+            let selectedPort: number = await getPort({port: getPort.makeRange(this.bindPort, this.bindPort + 500)});
+            this.app.listen(selectedPort);
+            logger.log(`Please browse to http://localhost:${selectedPort} to begin the configurations of the Repeater`);
+        })();
+    }
+
+    private async testAuthConnection(): Promise<boolean> {
+        const url: URL = new URL(this.authTestEndpoint);
+        const tokens: Tokens = this.readTokens();
+        return new Promise<boolean>((resolve) => {
+            let req: httpReq.Request = httpReq.post({
+                    url: url,
+                    form: {
+                        username: tokens.repeaterId, 
+                        password: tokens.authToken, 
+                    }
+                }, (error: any, response: httpReq.Response, body: string) =>{
+                    if (error || response.statusCode != 200) {
+                        resolve(false);
+                    }
+                    resolve(body === 'allow');
+                }
+            );
+            req.on('error', ()=> {
+                logger.error('Auth HTTP connection failed. Could not make HTTP call to auth endpoint.');
+                resolve(false);
+            });
+            setTimeout(() => {
+                logger.debug('testAuthConnection reached timeout. Destroying the HTTP connecttion.');
+                req.destroy();
+            }, this.processLaunchTimeout);
+        });
+    }
+
+    private async testHTTPConnection(): Promise<boolean> {
+        const url: URL = new URL(this.httpTestEndpoint);
+        const timeoutMs = 30 * 1000; // 30 seconds
+        return new Promise<boolean>((resolve) => {
+            let req: ClientRequest = https.get(url, (res: IncomingMessage) =>{
+                res.on('data', ()=>{
+                    logger.debug('Http connectivity test - received data the connection.The connection is succesfull.');
+                    resolve(true);
+                });
+                res.on('end', ()=> {
+                    logger.debug('Http connectivity test - connection closed by server. The connection is succesfull.');
+                    resolve(true);
+                });
```

Comment:
You don't need to parse the response body. Remove handlers, but don't forget to drain stream.